services/brig/src/Brig/IO/Intra.hs

```diff
@@ -979,7 +980,14 @@ guardLegalhold protectee userClients = do
   res <- lift . wrapHttp $ galleyRequest PUT req
   case Bilge.statusCode res of
     200 -> pure ()
-    403 -> throwE ClientMissingLegalholdConsent
+    403 -> case Bilge.responseJsonMaybe @Value res >>= (^? key "label") of
+      Just "missing-legalhold-consent" -> throwE ClientMissingLegalholdConsent
+      Just "missing-legalhold-consent-old-clients" -> throwE ClientMissingLegalholdConsentOldClients
+      _ ->
+        -- only happens if galley misbehaves (fisx: this could also be a parse error if we
+        -- used a more constraining type to send back & forth between brig and galley, but
+        -- merging brig and galley would make this train of thought go away more naturally).
+        throwE ClientMissingLegalholdConsent
```

Comment:
why can we throw this here? it appears this is more "undefined behavior", no?

Reply:
did you read the comment above?  i'm doing this because i don't want to run into an `undefined` here in case my understanding of the code is wrong.  this exception will be much more gentle to the user and chances are it'll be accurate "enough".